src/download_images.py

```diff
@@ -94,7 +87,10 @@ def get_image_from_coordinates(self, latitude: int, longitude: int) -> dict:
         results["image_lon"] = image["geometry"]["coordinates"][0]
         results["residual"] = closest_distance.m
         image_url = image["thumb_original_url"]
-        results["image_path"] = self._download_image(image_url, results["image_id"])
+        try:
+            results["image_path"] = self._download_image(image_url, results["image_id"])
+        except HTTPError or RetryError as e:
+            results["image_path"] = e.__class__.__name__
```

Comment:
I'm not sure about putting `HttpError` into the `image_path` field. I'm wondering if it would be better to add this as an extra column. It feels like it may be confusing for `image_path` to have magic values (which might be `HttpError`, `RetryError`, or something more mysterious if a subclass is thrown) that a user needs to know to discard. It would probably be better for those values to just be null. I recognize that this implementation is very compact, but I think it's worth trading off the compactness for more explicit and clear data.

Reply:
I do agree that having an error column is a better approach, but one piece of info that the current implementation has is where the error occurred. If the error is in the `image_id` column, then there was an error in the initial Mapillary API request to get images for a lat and long. If the error is in the `image_path` column, then there was an error in downloading the actual image. I'll try to find a method that both uses a new column for errors and has info about which step in the process it failed on.